### PR TITLE
[SofaGraphComponent] Fix a crashing bug in SceneCheckAPIChange.

### LINKED ISSUE #491: [SofaGraphComponent] Crash caused by check in SceneCheckAPIChanges when using RestShapeSpringsForceField
When trying to use the RestShapeSpringsForceField component (on branch master), runSofa crashes upon checking the scene. The responsible check tries to find a data called "external_rest_shape". However, that data has become a link (since after v17.06), so the function findData returns a NULL pointer.

It looks like that check cannot fulfill its purpose since the data will always be a link, no matter what the user wrote in their scene. It would need to parse the input string to verify whether the "@" is there in front of the attribute.

______________________________________________________

Suggested labels:

### PR DESCRIPTION
**Pull request by damienmarchal**
When a message does not have an 'suggestion' text then sofa will
crash. Because of Index out of bound.
This PR fix that.
Fixes #491 

______________________________________________________

This PR: 
- [ ] builds with SUCCESS for all platforms on the CI.
- [ ] does not generate new warnings.
- [ ] does not generate new unit test failures.
- [ ] does not generate new scene test failures.
- [x] does not break API compatibility.
- [x] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

File: modules/SofaGraphComponent/SceneCheckAPIChanges.cpp [hunk 1]
```diff
@@ -112,7 +113,7 @@ void SceneCheckAPIChange::doCheckOn(Node* node)
 void SceneCheckAPIChange::installDefaultChangeSets()
 {
     addHookInChangeSet("17.06", [](Base* o){
-        if(o->getClassName() == "RestShapeSpringsForceField" && o->findData("external_rest_shape")->isSet())
+        if(o->getClassName() == "RestShapeSpringsForceField" && o->findLink("external_rest_shape")->getSize() != 0)
```
Comment (guparan) [inline modules/SofaGraphComponent/SceneCheckAPIChanges.cpp]: This is messing with my brain again ><
What happens if there is no `external_rest_shape` parameter in my RestShapeSpringsForceField?

Comment (damienmarchal) [reply modules/SofaGraphComponent/SceneCheckAPIChanges.cpp]: A suspect a lot of bad things happens if you start removing the default Data field and links at runtime. 
In this case it crash. I fully agree to add a test condition and wonder why it was not done in the initial merged pR.

Comment (damienmarchal) [reply modules/SofaGraphComponent/SceneCheckAPIChanges.cpp]: @guparan If you have the time to add the test do it...I'm fully busy until my presentation at Toulouse thursday.

Comment (damienmarchal) [reply modules/SofaGraphComponent/SceneCheckAPIChanges.cpp]: Well, finally I started qtcreator and I was on the right branch...so made the fix.

File: modules/SofaGraphComponent/SceneCheckAPIChanges.cpp [hunk 2]
```diff
@@ -110,8 +113,11 @@ void SceneCheckAPIChange::doCheckOn(Node* node)
 void SceneCheckAPIChange::installDefaultChangeSets()
 {
     addHookInChangeSet("17.06", [](Base* o){
-        if(o->getClassName() == "RestShapeSpringsForceField" && o->findData("external_rest_shape")->isSet())
-            msg_warning(o) << "RestShapeSpringsForceField have changed since 17.06. The parameter 'external_rest_shape' is now a Link. To fix your scene you need to add and '@' in front of the provided path. See PR#315" ;
+        if(o->getClassName() == "RestShapeSpringsForceField" && o->findLink("external_rest_shape") )
+        {
+            if( o->findLink("external_rest_shape")->getSize() != 0)
```
Comment (guparan) [inline modules/SofaGraphComponent/SceneCheckAPIChanges.cpp]: What is `getSize() != 0` supposed to check? 
If a `RestShapeSpringsForceField` is implemented with a wrong `external_rest_shape`, the link will be created but empty. If a `RestShapeSpringsForceField` is implemented with NO `external_rest_shape`, the link will still be created and empty.
I I feel like there is no way to differentiate a wrong Link from an empty Link after component instantiation.

**Pull request merged**